src/crypto/mod.rs

```diff
@@ -103,3 +105,68 @@ pub trait Cryptographer: Send + Sync + 'static {
     ) -> Result<Vec<u8>>;
     fn random_bytes(&self, dest: &mut [u8]) -> Result<()>;
 }
+
+/// Run a small suite of tests to check that a `Cryptographer` backend is working correctly.
+///
+/// You should only use this is you're implementing a custom `Cryptographer` and want to check
+/// that it is working as intended. This function will panic if the tests fail.
+///
```

Comment:
The disadvantage of hiding things from the public API, is that it makes it harder to test our custom NSS-based ece backend over in the application-services repo. It currently [has its own copy of all the aes128gcm tests](https://github.com/mozilla/application-services/blob/02bf1a0b7c951896e715a5b00d43d5af80bfe479/components/support/rc_crypto/src/ece_crypto.rs#L178) which use a bunch of the now-hidden implementation details. What if instead, we exposed a function here specifically for testing custom backends?

(My theory with making it generic over `T: Cryptographer` rather than e.g. taking a `Box<dyn Cryptographer>` is that it won't produce any code unless you actually try to use a concrete instantiation of it, and since we only use it in tests, it won't contribute to the size of our production builds. I have done exactly zero work to confirm whether that's actually what will happen).

Reply:
Ref https://github.com/mozilla/application-services/pull/3941 for the corresponding appservices change.

Reply:
IIRC: you might be able to use a feature flag to make a "test" version that does expose this as public. It's a bit hacky, but we've done something similar in other repos.